src/audio/drc/drc.c

```diff
@@ -68,7 +68,7 @@ inline int drc_init_pre_delay_buffers(struct drc_state *state,
 				      size_t sample_bytes,
 				      int channels)
 {
-	size_t bytes_per_channel = sample_bytes * DRC_MAX_PRE_DELAY_FRAMES;
+	size_t bytes_per_channel = sample_bytes * CONFIG_DRC_MAX_PRE_DELAY_FRAMES;
```

Comment:
have you tried with rballoc? That might be way you are hitting ENOMEM

Reply:
You are absolutely right!! I just knew that there are several memory heap zones to allocate from.
https://thesofproject.github.io/latest/api/memory-alloc-api.html

Reply:
Then I changed the default value of DRC_MAX_PRE_DELAY_FRAMES to 512. The pre-delay second is 0.006 in CRAS and the correspondent frame number is 288 (0.006*48000). The max frame number just needs to be larger than 288 and be a 2^N number, then we could guarantee the behavior is coherent as CRAS.